models/ncnewsmodels.js

```diff
@@ -34,26 +37,36 @@ exports.selectArticleById = (article_id) => {
       JOIN comments ON articles.article_id = comments.article_id
       GROUP BY articles.article_id
       ORDER BY created_at DESC;
-    `,
-    );
-    return db.query(selectArticlesQuery).then((result) => { 
-    return(result.rows)
-   });
-  };
+    `
+  );
+  return db.query(selectArticlesQuery).then((result) => {
+    return result.rows;
+  });
+};
 
-  exports.selectAllComments = (id) => {
-    const selectCommentsQuery = format(
-      `SELECT * FROM comments WHERE article_id = %L
-      ORDER BY created_at DESC;`, id
-    );
-    return db.query(selectCommentsQuery).then(({rows}) => { 
-      if (rows.length === 0) {
-      return rows
-    }
+exports.selectAllComments = (id) => {
+  const selectArticlesQuery = format(
+    `SELECT * FROM articles WHERE article_id = %L;`,
+    id
+  );
+  return db.query(selectArticlesQuery).then((result) => {
     
-      return rows;
-    });
-  };
-  
-
-
+    if (result.rows !== 0 && result.rows.length !== 0) {
+      const selectCommentsQuery = format(
+        `SELECT * FROM comments WHERE article_id = %L
+          ORDER BY created_at DESC;`,
+        id
+      );
+      return db.query(selectCommentsQuery).then((result) => {
+     
+        if (result.rows.length === 0) {
+          return result.rows;
+        } else {
+          return result.rows;
+        }
```

Comment:
You can just return result.rows here without the if checking the length now, you are returning the rows no matter if they are empty or not :)